server/src/main/java/org/apache/druid/segment/loading/StorageLocation.java

```diff
@@ -154,7 +166,8 @@ public synchronized File reserve(String segmentFilePathToAdd, String segmentId,
   {
     final File segmentFileToAdd = new File(path, segmentFilePathToAdd);
     if (files.contains(segmentFileToAdd)) {
-      return null;
+      //TODO: is this change ok?
+      return segmentFileToAdd;
```

Comment:
What's the reasoning for this change?

Reply:
a `null` result would imply that the reservation has actually failed because the storage location can't handle the file size and that's how it is interpreted outside. But if the file is already reserved, the operation didn't technically fail. With this change, the method can be called multiple times with correct semantics.

Reply:
What's the concern that the TODO is referencing?

Reply:
Your reasoning makes sense to me, but the callers currently depend on this behavior to avoid duplicate work. ​For example, [this line](https://github.com/apache/druid/pull/11398/files#diff-bf96197b1a1e78ae0c5ccc6e1e0757eccbabe9b0b5c91035a438462440443580R286) checks whether `storageDir` is null before downloading the segment. I think this method should return something that is indicative of whether the call reserves some space successfully or the call returns immediately at this line.